### PR TITLE
Do not make invalid android config with only NDK/SDK configured

### LINKED ISSUE #31843: Attempt to build with NDK only in configure.py results in invalid android.bzl
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04 (docker)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: -
- TensorFlow installed from (source or binary): Source
- TensorFlow version: 1.14.0
- Python version: 3.7
- Installed using virtualenv? pip? conda?: pip yes
- Bazel version (if compiling from source): 0.24.1
- GCC/Compiler version (if compiling from source): Clang from NDK r17b
- CUDA/cuDNN version: -
- GPU model and memory: -

**Describe the problem**
I'm attempting to build tensorflow light using android NDK compiler, without plugging in SDK.
Therefore I set `TF_SET_ANDROID_WORKSPACE` to 0, while I have `ANDROID_NDK_HOME` pointing to location of valid NDK installation.

Attempting to perform such build results in error:
```
ERROR: /root/.cache/bazel/_bazel_root/9deba62a2f90185d207a955a930b77dc/external/local_config_android/android.bzl:10:2: indentation error
ERROR: error loading package '': Extension 'android.bzl' has errors
ERROR: error loading package '': Extension 'android.bzl' has errors
```

This happens because configure.py adds ndk location (specified in env variable) by appending to android.bzl that has only pass in body of function.
But it is added with 2 spaces, while rest of android.bzl was with 4 tabs.

This is the first problem, I have to manually remove `pass` and leave only NDK setup
Probably there should not be such difference in tabulation when generating `android.bzl`

The second issue, is whether it should be possible to build `//tensorflow/lite:libtensorflowlite.so` with NDK only.
I was able to do so after manually correcting `android.bzl`

**Provide the exact sequence of commands / steps that you executed before running into the problem**

- Environment variables:

```
PYTHON_BIN_PATH = <valid location python>
USE_DEFAULT_PYTHON_LIB_PATH = '1'
TF_ENABLE_XLA = '0'
TF_NEED_ROCM = '0'
TF_NEED_CUDA = '0'
TF_NEED_MPI = '0'
TF_DOWNLOAD_CLANG = '0'
TF_SET_ANDROID_WORKSPACE = '0'
TF_CONFIGURE_IOS = '0'
ANDROID_NDK_HOME = <valid location of NDK>
ANDROID_NDK_API_LEVEL = '18'
TF_NEED_OPENCL_SYCL = '1'
TF_NEED_COMPUTECPP = '0'
TRISYCL_INCLUDE_DIR = '<path to cloned repo of TrySYCL>/include'
HOST_CXX_COMPILER = '<NDK path>/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++'
HOST_C_COMPILER = '<NDK path>/toolchains/llvm/prebuilt/linux-x86_64/bin/clang'
CC_OPT_FLAGS = '-Wno-sign-compare'
```

- Run `python configure.py`
- Rim `bazel shutdown`
- Run build with `bazel build --config=opt --config=v2 --cxxopt='--std=c++11' --define=no_tensorflow_py_deps=true --config=android_arm64 //tensorflow/lite:libtensorflowlite.so --verbose_failures`

**Any other info / logs**
Manual correction of `android.bzl` results in successful build, but it seems there are no symbols related to GPU delegate (I guess a separate target is required?)

### PR DESCRIPTION
**Pull request by DoumanAsh**
Fixes #31843

Corrects indent of templates for both SDK and NDK to use 4 spaces.
Makes `pass` statement only if both NDK and SDK are missing

### CODE REVIEW

**Review by frankchn — APPROVED** [review]

File: third_party/android/android_configure.bzl [hunk 1]
```diff
@@ -45,18 +45,21 @@ def _android_autoconf_impl(repository_ctx):
...
             build_tools_version,
         )
 
-    ndk_rule = "pass"
+    ndk_rule = ""
     if all([ndk_home, ndk_api_level]):
         ndk_rule = _ANDROID_NDK_REPO_TEMPLATE % (ndk_home, ndk_api_level)
 
+    if ndk_rule == "" and sdk_rule == "":
+        sdk_rule = "pass"
```
Comment (frankchn) [inline third_party/android/android_configure.bzl]: Actually why is this changed?

**Review by frankchn — CHANGES_REQUESTED** [review]

File: third_party/android/android_configure.bzl (hunk 1 above)
Comment (DoumanAsh) [reply third_party/android/android_configure.bzl]: If both SDK and NDK are missing, I decided to add just pass statement instead of nothing.
I guess it might not be necessary, and empty function would work too

**Review by frankchn — APPROVED** [review]

**Comment by DoumanAsh** [comment]
There is surprising CI failure:

```
[0 / 26] [Prepa] BazelWorkspaceStatusAction stable-status.txt
ERROR: /tmpfs/tmp/bazel/external/com_google_protobuf/BUILD:106:1: C++ compilation of rule '@com_google_protobuf//:protobuf_lite' failed (Exit 1) gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections ... (remaining 32 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
gcc: error: unrecognized command line option '-std=c++14'
```
